### DISCUSSION TITLE
https://theme-hope.vuejs.press/zh/guide/interface/darkmode.html

### DISCUSSION THREAD

**Discussion by giscus** [vuepress-theme-hope/giscus-discussions#676] [Announcements]
# https://theme-hope.vuejs.press/zh/guide/interface/darkmode.html

在深色模式下，页面会使用深色背景与浅色文字，以让你更加舒适。

https://theme-hope.vuejs.press/zh/guide/interface/darkmode.html

**Comment by jupiter911** [comment]
请问有没有更详细一些的说明，比如这个代码应该写在什么文件里的什么位置？

**Reply by Mister-Hope** [reply]
https://theme-hope.vuejs.press/zh/config/intro.html#配置概念

**Comment by superen7527** [comment]
请问如何根据主题切换不同的图标或者给图标上色 只看到切换主题色的写法.找了一个多小时没找到相关说明 我用的是本地图标 希望可以指点一下 谢谢

**Reply by superen7527** [reply]
问gpt也没得到答案

**Reply by firstmeet1108** [reply]
主题颜色的值被定义在页面的html元素上，可以用MutationObserver监听html元素的data-theme属性变动实现扩展内容的主题颜色切换。

**Reply by raysun0501** [reply]
使用css属性选择器监听<html>标签上data-theme属性的值，然后根据值的变化修改css样式。是可以实现的，详细情况可以问问GPT4.

**Comment by FSWei** [comment]
import { useDarkmode } from "vuepress-theme-hope/client";

这个目前还能用吗？为啥我会报错：
模块“"vuepress-theme-hope/client"”没有导出的成员“Darkmode”。

**Reply by Mister-Hope** [reply]
所以你想告诉我你导入了useDarkmode，但报错是主题没有导出 Darkmode?

**Reply by FSWei** [reply]
不好意思，刚刚复制错了，是：
模块“"vuepress-theme-hope/client"”没有导出的成员“useDarkmode”。

**Reply by FSWei** [reply]
我是在navbar.ts里导入的，这有影响吗

**Reply by Mister-Hope** [reply]
导航栏是一个node端的配置，哪来的夜间模式信息？

**Reply by FSWei** [reply]
哦哦，原来如此，谢谢
